src/CFamily/Analysis/CLangAnalyzer.cs

```diff
@@ -1,72 +0,0 @@
-﻿/*
- * SonarLint for Visual Studio
- * Copyright (C) 2016-2024 SonarSource SA
- * mailto:info AT sonarsource DOT com
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 3 of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- */
-
-using System.ComponentModel.Composition;
-using System.Diagnostics.CodeAnalysis;
-using SonarLint.VisualStudio.Core.Analysis;
-
-namespace SonarLint.VisualStudio.CFamily.Analysis
-{
-    internal interface ICFamilyAnalyzer : IAnalyzer
-    {
-        void ExecuteAnalysis(string path,
-            IEnumerable<AnalysisLanguage> detectedLanguages,
-            IIssueConsumer consumer,
-            IAnalyzerOptions analyzerOptions,
-            IAnalysisStatusNotifier statusNotifier,
-            CancellationToken cancellationToken);
-    }
-
-    [ExcludeFromCodeCoverage]
-    [Export(typeof(ICFamilyAnalyzer))]
-    [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class CLangAnalyzer : ICFamilyAnalyzer
-    {
-        [ImportingConstructor]
-        public CLangAnalyzer()
-        {
-        }
-
-        public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
-        {
-            return languages.Contains(AnalysisLanguage.CFamily);
-        }
-
-        public void ExecuteAnalysis(
```

Comment:
It seems that issue streaming also breaks after SLOOP. Consider the following example code (need to configure the project as c++20):

```
void fun(int* p) {
	if (p == 0) {
		*p = 42;
	}
#define N 14
	static const int s[] = { [1 << N] = 0 };
}
```
Before SLOOP, issues used to be reported as soon as they are detected in the analyzer, so the list of issues would start by reporting the unused variable `s`, and after a few seconds, we get the null dereference report in the assignment `*p = 42`.

OTOH, after SLOOP, all issues disappear when an analysis is triggered, and they are reported only in a single shot after a few seconds.

Do you know what could be the reason behind this? I am also happy to demonstrate the behavior in a call if it makes things faster...

Reply:
Will be handled in a separate PR https://sonarsource.atlassian.net/browse/SLVS-1679